src/d_main.c

```diff
@@ -255,16 +255,9 @@ void D_Display (void)
     case GS_LEVEL:
       if (!gametic)
         break;
-      if (automapactive)
```

Comment:
I guess most of the changes below could be spared if you changed this to `automapactive && !automapoverlay`.